feat(Label): penta updates

kmcfaul opened the pull request:
<!-- What changes are being made? Please link the issue being addressed. -->
**What**: Closes #9981

Added 'orangered' color option & added examples.
Added `isAddLabel` flag, intending for label group's add button.
Added `pf-m-clickable` to non-overflow and non-add label buttons or links.
Added `pf-m-no-padding` to close buttons.
Updated filled, outline, and compact examples to use css for layout instead of manual whitespace.

patternfly-build commented:
Preview: https://patternfly-react-pr-10037.surge.sh

mcoker reviewed (approved):
LGTM!

Just a heads up, I have another [PR up in core](https://github.com/patternfly/patternfly/pull/6227) to add status variants, and adds a "filled" modifier to the default/filled labels. It also replaces the info-circle icons we're using as the example icon in labels with a cube icon since the info-ircle icon is used with info labels.

thatblindgeye reviewed (approved):
Looks good. Had a couple comments below, but they aren't blocking to this PR.

- @mcoker the editable label cursor style remains a pointer when the label is currently editable. I'd expect the cursor to be `text` when editing a label (this would be a core followup, unless the intent was to have the pointer cursor)
- Can we open a followup issue to add status support/update non-status labels in example to use the icon Michael mentioned above

kmcfaul commented:
Opened #10047 to follow up for status labels, in case it doesn't get in on this PR.

kmcfaul commented:
- Removed `isOverflowLabel` and `isAddLabel`
- Added `overflow` and `add` to `variants`

Since these label types seem to be mutually exclusive with filled and outline. LMK if either should be a flag still instead.

lboehling commented:
lgtm!

tlabaj commented on packages/react-core/src/components/Label/Label.tsx:
```diff
@@ -58,8 +58,6 @@ export interface LabelProps extends React.HTMLProps<HTMLSpanElement> {
   /** Href for a label that is a link. If present, the label will change to an anchor element. This should not be passed in if the onClick prop is also passed in. */
   href?: string;
-  /** Flag indicating if the label is an overflow label. */
-  isOverflowLabel?: boolean;
```
Can you please opne code mode issue for this removed prop.

tlabaj reviewed (approved):
LGTM

patternfly-build commented:
Your changes have been released in: 
 - @patternfly/react-code-editor@6.0.0-alpha.22
 - @patternfly/react-core@6.0.0-alpha.22
 - @patternfly/react-docs@7.0.0-alpha.23
 - @patternfly/react-drag-drop@6.0.0-alpha.3
 - demo-app-ts@5.1.1-alpha.21
 - @patternfly/react-table@6.0.0-alpha.22

Thanks for your contribution! :tada:
